omnistrate_tests/test_tls_expiration.py

```
@@ -0,0 +1,516 @@
+        """
+        try:
+            logger.debug("Checking certificate for %s:%d", hostname, port)
+
+            # Create SSL context
+            context = ssl.create_default_context()
+            context.check_hostname = False
+            context.verify_mode = ssl.CERT_REQUIRED
+
+            # Connect and get certificate
+            with socket.create_connection((hostname, port), timeout=timeout) as sock:
+                with context.wrap_socket(sock, server_hostname=hostname) as ssock:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)